sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -95,6 +95,14 @@ case class DataSource(
   lazy val sourceInfo: SourceInfo = sourceSchema()
   private val caseInsensitiveOptions = CaseInsensitiveMap(options)
   private val equality = sparkSession.sessionState.conf.resolver
+  // The operations below are expensive therefore try not to do them if we don't need to, e.g.,
+  // in streaming mode, we have already inferred and registered partition columns, we will
+  // never have to materialize the lazy val below
+  private lazy val tempFileIndex = {
```

Comment:
it's only used once, no need to be a lazy val, we can just inline it.

Reply:
I moved it here on purpose. So it may be avoid being created twice in the future.
I am OK to inline it.

Reply:
let's just inline it. People can still create a new index in the future, technically this can't prevent users from doing that.